### PR TITLE
Fix cursed mask bug

### PR DESCRIPTION
**Pull request by FireNameFN**
## About the PR
Fixed bug where you can equip cursed mask that makes you angry, die, get rid of cursed mask and get healed. With this bug after these actions you will be `notarget` for almost all NPCs. But with this fix behavior will be expectable.

## Why / Balance

Bug fix.

## Technical details
Save copy of faction list.

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

No changelog.

### CODE REVIEW

**Comment by deltanedas** [comment]
does HashSet not have Clone or something this seems like it would trip up anyone seeing it in the future not familiar with why its being used

File: Content.Server/Clothing/Systems/CursedMaskSystem.cs [hunk 1]
```diff
@@ -51,7 +51,7 @@ protected override void TryTakeover(Entity<CursedMaskComponent> ent, EntityUid w
         }
 
         var npcFaction = EnsureComp<NpcFactionMemberComponent>(wearer);
-        ent.Comp.OldFactions = npcFaction.Factions;
+        ent.Comp.OldFactions = [.. npcFaction.Factions];
         _npcFaction.ClearFactions((wearer, npcFaction), false);
```
Comment (metalgearsloth) [inline Content.Server/Clothing/Systems/CursedMaskSystem.cs:55]: Just use Clear + UnionWith, less allocations.

**Review by slarticodefast — APPROVED** [review]
I tested it and it fixes the issue as intended.
Thank you for your contribution.

**Comment by chromiumboy** [comment]
Yup, this issue seems to be resolved, thanks for the fix and for your first contribution

**Pull request merged**